edgedb/retry.py

```diff
@@ -5,23 +5,37 @@
 from . import transaction as _transaction
 
 
-class AsyncIOIteration(_transaction.AsyncIOTransaction):
+class AsyncIOIteration(_transaction.BaseAsyncIOTransaction):
     def __init__(self, retry, owner, iteration):
         super().__init__(owner, retry._options.transaction_options)
         self.__retry = retry
         self.__iteration = iteration
+        self.__started = False
 
-    async def start(self):
+    async def _ensure_transaction(self):
         if not self._managed:
             raise errors.InterfaceError(
                 "Only managed retriable transactions are supported. "
                 "Use `async with transaction:`"
             )
-        await self._start(single_connect=self.__iteration != 0)
+        if not self.__started:
+            self.__started = True
+            await self._start(single_connect=self.__iteration != 0)
+
+    async def __aenter__(self):
+        self._managed = True
```

Comment:
Can we add some code to prevent entering the same transaction object twice?

Reply:
Sure thing!